dotCMS/src/main/java/com/dotcms/rest/ContentResource.java

```diff
@@ -1081,22 +1081,34 @@ protected void processMap(Contentlet contentlet, Map<String,Object> map) throws
 
 				// look for relationships
 				Map<Relationship,List<Contentlet>> relationships=new HashMap<Relationship,List<Contentlet>>();
-				for(Relationship rel : FactoryLocator.getRelationshipFactory().byContentType(st)) {
-					String relname=rel.getRelationTypeValue();
-					String query=(String)map.get(relname);
-					if(UtilMethods.isSet(query)) {
-						try {
-							List<Contentlet> cons=APILocator.getContentletAPI().search(
-									query, 0, 0, null, APILocator.getUserAPI().getSystemUser(), false);
-							if(cons.size()>0) {
-								relationships.put(rel, cons);
-							}
-							Logger.info(this, "got "+cons.size()+" related contents");
-						} catch (Exception e) {
-							Logger.warn(this, e.getMessage(), e);
-						}
-					}
+				List<Relationship> rels = FactoryLocator.getRelationshipFactory().byContentType(st);
+				for(Relationship rel : rels) {
+				    String relname=rel.getRelationTypeValue();
+				    String query=(String)map.get(relname);
+				    if(UtilMethods.isSet(query)) {
+				        try {
+				            List<Contentlet> cons=APILocator.getContentletAPI().search(
+				                    query, 0, 0, null, APILocator.getUserAPI().getSystemUser(), false);
+				            if(cons.size()>0) {
+				                relationships.put(rel, cons);
+				            }
+				            Logger.info(this, "got "+cons.size()+" related contents");
+				        } catch (Exception e) {
+				            Logger.warn(this, e.getMessage(), e);
+				        }
+				    } else {
+			            if(!relationships.containsKey(rel)){
+			                relationships.put(rel, new ArrayList<Contentlet>());
+			            }
+			            List<Contentlet> cons = APILocator.getContentletAPI().getRelatedContent(
+			                    contentlet, rel, APILocator.systemUser(), false);
+			            for (Contentlet c : cons) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)